packages/next/client/index.tsx

```diff
@@ -508,11 +510,13 @@ function renderReactElement(
   const reactEl = fn(shouldHydrate ? markHydrateComplete : markRenderComplete)
   if (process.env.__NEXT_REACT_ROOT) {
     if (!reactRoot) {
-      reactRoot = (ReactDOM as any).createRoot(domEl, {
-        hydrate: shouldHydrate,
-      })
+      if (shouldHydrate && typeof ReactDOM18.hydrateRoot === 'function') {
+        reactRoot = ReactDOM18.hydrateRoot(domEl, reactEl)
```

Comment:
The reactRoot would always be initialized through `hydrateRoot` right? We don't have to maintain backwards compatibility with React 18 alpha releases